platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherProbeRegistry.java

```diff
@@ -124,31 +199,44 @@ public enum State {
             TRIGGERED
         }
 
-        private final File watchableHierarchy;
+        private final Set<File> watchableHierarchies;
         private final File probeFile;
+        private final boolean subPathOfWatchableHierarchy;
         private State state = State.UNARMED;
 
-        public WatchProbe(File watchableHierarchy, File probeFile) {
-            this.watchableHierarchy = watchableHierarchy;
+        public WatchProbeImpl(File probeFile, boolean subPathOfWatchableHierarchy) {
             this.probeFile = probeFile;
+            this.subPathOfWatchableHierarchy = subPathOfWatchableHierarchy;
+            this.watchableHierarchies = new HashSet<>();
+        }
+
+        public File getLocation() {
+            return probeFile.getParentFile();
+        }
+
+        public boolean isSubPathOfWatchableHierarchy() {
+            return subPathOfWatchableHierarchy;
         }
 
-        public synchronized void arm() throws IOException {
+        public synchronized void arm() {
             switch (state) {
                 case UNARMED:
                     state = State.ARMED;
                     //noinspection ResultOfMethodCallIgnored
                     probeFile.getParentFile().mkdirs();
                     try (FileOutputStream out = new FileOutputStream(probeFile)) {
                         out.write(Longs.toByteArray(System.currentTimeMillis()));
+                    } catch (IOException e) {
+                        LOGGER.debug("Could not arm watch probe for hierarchies: {}", watchableHierarchies, e);
+                        return;
```

Comment:
❓ Why this change?

Reply:
It's part of refactoring, it was there before.
https://github.com/gradle/gradle/blob/a23d1dafbfcff1767a2b94a1c6f3d8464e7d3489/platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherProbeRegistry.java#L67-L72